unity-renderer/Assets/Rendering/ProceduralSkybox/ToolProceduralSkybox/Scripts/SkyboxController.cs

```diff
@@ -0,0 +1,573 @@
+using System;
+using System.Linq;
+using UnityEngine;
+using DCL.ServerTime;
+
+namespace DCL.Skybox
+{
+    /// <summary>
+    /// This class will handle runtime execution of skybox cycle. 
+    /// Load and assign material to the Skybox.
+    /// This will mostly increment the time cycle and apply values from configuration to the material.
+    /// </summary>
+    public class SkyboxController : IPlugin
+    {
+        public event SkyboxConfiguration.TimelineEvents OnTimelineEvent;
+
+        public static SkyboxController i { get; private set; }
+
+        public const string DEFAULT_SKYBOX_ID = "Generic_Skybox";
+
+        public string loadedConfig;
+        //Time for one complete circle. In Hours. default 24
+        public float cycleTime = 24;
+        public float lifecycleDuration = 2;
+
+        private float timeOfTheDay;                            // (Nishant.K) Time will be provided from outside, So remove this variable
+        private Light directionalLight;
+        private SkyboxConfiguration configuration;
+        private Material selectedMat;
+        private bool overrideDefaultSkybox;
+        private string overrideSkyboxID;
+        private bool isPaused;
+        private float timeNormalizationFactor;
+        private int slotCount;
+        private bool overrideByEditor = false;
+
+        // Reflection probe
+        private ReflectionProbe skyboxProbe;
+        bool probeParented = false;
+        private float reflectionUpdateTime = 1;                                 // In Mins
+        private ReflectionProbeRuntime runtimeReflectionObj;
+
+        // Timer sync
+        private int syncCounter = 0;
+        private int syncAfterCount = 10;
+
+        public SkyboxController()
+        {
+            i = this;
+
+            // Find and delete test directional light obj if any
+            Light[] testDirectionalLight = GameObject.FindObjectsOfType<Light>().Where(s => s.name == "The Sun_Temp").ToArray();
+            for (int i = 0; i < testDirectionalLight.Length; i++)
+            {
+                GameObject.DestroyImmediate(testDirectionalLight[i].gameObject);
+            }
```

Comment:
This code looks like a temporary workaround, we should remove it.

Reply:
while using the tool, a temproary directional light is created by the tool (in case a directional light is not present). This code just remove any temporary light created by the tool so that there should not be multiple directional lights